cli/azd/pkg/environment/environment.go

```diff
@@ -70,19 +70,41 @@ func FromFile(file string) (Environment, error) {
 	return env, nil
 }
 
-func GetEnvironment(azdContext *azdcontext.AzdContext, name string) (Environment, error) {
+func GetEnvironment(azdContext *azdcontext.AzdContext, name string) (*Environment, error) {
```

Comment:
Would it be worth it to move from struct to interface to represent the Environment?

Same way we do for `AzCli`, should we define just the Environment's expected behavior and provide one implementation with the current approach (memory map + optional file)

This would make it easier to test and would start setting a pattern here, for preferring interfaces over direct components

Reply:
As I continue to refactor `Environment`, I will think about this.  The ergonomics end up being a little worse because you can't have fields on the interface, but it may be reasonable to do longer term.

The other option would be to factor out the "persistence" logic into an interface by have `Environment` itself remain a struct.  By default we could use the current file based logic, but ephemeral environments could be implemented by providing a no-op implementation.  I slightly prefer that design approach instead of having to implement a bunch of the same "business logic" across multiple environment implementations.